src/components/stack-chart/Canvas.js

```diff
@@ -345,6 +348,13 @@ class StackChartCanvasImpl extends React.PureComponent<Props> {
             intW + BORDER_OPACITY,
             intH
           );
+
+          if (searchStringsRegExp && searchStringsRegExp.test(text)) {
```

Comment:
Because the searchStringsRegExp is shared, it needs to be "reset" when using it. I think this is the source of the bug I found. You can do this: https://github.com/firefox-devtools/profiler/blob/510a546b4f4903d5987c575ace0fb5a50adadaf2/src/components/shared/TreeView.js#L77-L78